_layouts/post.html

```diff
@@ -16,9 +16,21 @@
   {% include header.html %}
 
   <main>
-    <div class="post">
-      <h1>{{ page.title }}</h1>
+    <div class="post-header">
+      {% if page.header-image %}
+        <img class="post-header-image" src="{{ page.header-image | relative_url }}">
+      {% elsif page.thumbnail %}
+        <img class="post-header-image" src="{{ page.thumbnail | relative_url }}">
+      {% endif %}
+    </div>
 
+    {% if page.header-image or page.thumbnail %}
+      <div class="post post-body-header-border">
+    {% else %}
+      <div class="post">
+    {% endif %}
+      <h1 class="post-title">{{ page.title }}</h1>
```

Comment:
Indentation here is unclear (this also applies for through the rest of the patch). For consistency for now, I'd follow the same style as used in `_layouts/page.html`.
The style there's not perfect and maybe we need to change that at some point, but probably as a different discussion under a different issue/pr.